src/parser.yy

```diff
@@ -709,6 +716,19 @@ offsetof_expr:
         |       OFFSETOF "(" expr "," struct_field ")"             { $$ = driver.ctx.make_node<ast::Offsetof>($3, $5, @$); }
                 ;
 
+typeok_expr:
+                TYPEOK "(" any_type ")"  { $$ = driver.ctx.make_node<ast::Typeok>($3, @$); }
+                ;
+
+typeof_expr:
+                TYPEOF "(" type ")"  { $$ = driver.ctx.make_node<ast::Typeof>($3, @$); }
+        |       TYPEOF "(" expr ")"  { $$ = driver.ctx.make_node<ast::Typeof>($3, @$); }
+                ;
+
+any_type:
```

Comment:
naming bikeshedding: `derived_type`?

Reply:
I used "any" because it is either:

- A concrete type-name
- A `typeof` expression

Isn't `derived_type` specified to the second one?

Reply:
yeah, I guess `derived_type` is specific to the second one... ok, I don't have a better name than `any_type` ATM